spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/Sample.scala

```diff
@@ -132,8 +143,187 @@ class TensorSample[T: ClassTag](
   }
 }
 
+/**
```

Comment:
I don't think we need multiple type of sample. For storage save purpose, we can have one single Array to store all the information. Array need less storage than (Tensor, T). Another benifit is you need not write so many sampleToBatch.

@jason-dai